doc-surrealdb_versioned_docs/version-1.x/reference-guide/vector-search.mdx

```diff
@@ -0,0 +1,219 @@
+  className="lead screenshot"
+  sources={{
+    light: useBaseUrl('img/light/distance-metrics.png'),
+    dark: useBaseUrl('img/distance-metrics.png'),
+  }}
+/>
+
+The choice of distance or similarity function depends on the nature of your data and the specific requirements of your application.
+
+In the previous example of actors' facial data, we assumed that the embeddings represented the facial characteristics. We used the `vector::similarity::cosine` function because cosine similarity is typically preferred when absolute distances are less important, but proportions and direction matter more. 
+This means that, in our use case, we are less concerned with two noses being the same size and more interested in two faces having similar proportions.
+As cosine similarity measures the orientation of the vectors, rather than their absolute magnitudes, this makes it suitable to help capture the overall similarity in facial features and proportions, while being less sensitive to factors like scale or image brightness.
+
+Now, if your embeddings represented the actors’ careers, their awards, and their box office revenue, and we were trying to find who was the most successful actor after you or your favorite actor, using Euclidean distance would make more sense. 
+
+```surql
+SELECT id, vector::distance::euclidean(embedding, $person_embedding) AS dist FROM actor WHERE embedding <|1|> $person_embedding;
+```
+
+## Vector Indexes
+
+When it comes to search, you can always use brute force. 
+In SurrealDB, you can use the [Brute force approach](/docs/surrealdb/surrealql/statements/define/indexes#brute-force-method) to search through your vector embeddings and data. 
+
+Brute force search compares a query vector against all vectors in the dataset to find the closest match. As this is a brute-force approach, you do not create an index for this approach. 
+
+The brute force approach for finding the nearest neighbour is generally preferred in the following use cases:
+
+- Small Datasets / Limited Query vectors: For applications with small datasets, the overhead of building and maintaining an index might outweigh its benefits. In such cases, the brute force approach is optimal.
+- Guaranteed Accuracy: Since the brute force method compares the query vector against every vector in the dataset, it guarantees finding the exact nearest vectors based on the chosen distance metric (like Euclidean, Manhattan, etc.).
+- Benchmarking Models: The Brute force approach can be used as a reference, and help benchmark the performance of other approximate alternatives like HNSW
+
+While brute force can give you exact results, it's computationally expensive for large datasets. 
+
+In most cases, you do not need a 100% exact match, and you can give it up for faster, high-dimensional searches to find the approximate nearest neighbour to a query vector. 
+
+This is where Vector indexes come in. 
+
+In SurrealDB, you can perform a vector search using the two primary indexes:
```

Comment:
```suggestion
In SurrealDB, you can perform a vector search using the two secondary indexes:
```

The term `primary` is reserved for the ID of the record, which is the index of the key-value store himself. Every other indexes (unique, standard, vector search, fts) are secondary indexes.

Reply:
Ahh. I meant 2 main indexes. I can see how it might confuse the reader between primary and secondary. Thank you for pointing this out.